src/Sentry.Unity.Android/SentryJava.cs

```diff
@@ -62,7 +65,11 @@ public SentryJava(IDiagnosticLogger? logger, IAndroidJNI? androidJNI = null)
 
     public bool? IsEnabled()
     {
-        HandleJniThreadAttachment();
+        if (!MainThreadData.IsMainThread())
+        {
+            _logger?.LogError("Calling IsEnabled() on Android SDK requires running on MainThread");
+            return null;
+        }
```

Comment:
This is called by the SDK as part of the initialization process which is currently explicitly required to run on the main thread. I'm adding the check here now too so any future changes to the initialization flow (i.e. programmatic through user code) is not turning into a footgun.